pkg/reconciler/route/reconcile_resources.go

```diff
@@ -240,3 +240,51 @@ func (c *Reconciler) reconcileTargetRevisions(ctx context.Context, t *traffic.Co
 	}
 	return eg.Wait()
 }
+
+func (c *Reconciler) reconcileCertificates(ctx context.Context, r *v1alpha1.Route, desiredCerts []*netv1alpha1.Certificate) ([]*netv1alpha1.Certificate, error) {
+	result := []*netv1alpha1.Certificate{}
+	gvk := netv1alpha1.SchemeGroupVersion.WithKind("Certificate")
+	for _, desiredCert := range desiredCerts {
+		c.tracker.Track(objectRef(desiredCert, gvk), r)
+		cert, err := c.reconcileCert(ctx, r, desiredCert)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, cert)
+	}
+	return result, nil
```

Comment:
How are certs cleaned up?  Given quotas, every Certificate we leave around is going to consume some of that quota in perpetuity.

Reply:
Now we add the OwnersReferences on each non-wildcard `Certificate`. So `Certificates` of a Route will be cleaned up when deleting a `Route`.
But I think this GC strategy may be a bit aggressive as Let'sEncryt has quota of issuing 5 duplicate certificates per week (https://letsencrypt.org/docs/rate-limits/). So if user deletes and then recreates a same ksvc for 5 times in a week, the duplicate certificate quota could be ran out.
I think a bit more moderate GC strategy is to cache the Certificate for a while (1 day probably) after its Route is deleted so the Certificate could be reused in the future. I can work on a separate PR for this GC strategy after this PR is checked in. WDYT? :)